lib/liquid/context.rb

```diff
@@ -189,7 +189,7 @@ def find_variable(key)
       if scope.nil?
         @environments.each do |e|
           variable = lookup_and_evaluate(e, key)
```

Comment:
strict_variables really shouldn't be raising in lookup_and_evaluate or we should be calling something else here, since this won't work properly if `@evironments` has a length of more than 1.

For example:

```ruby
template = Liquid::Template.parse("{{ foo }}")
template.assigns["foo"] = "works"
puts template.render!({}).inspect  # => "works"
puts template.render!({}, strict_variables: true).inspect  # raises Liquid::UndefinedVariable
```